python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -0,0 +1,758 @@
+                            self.get_actions(data)
+                            return
+                    else:
+                        self.logger.debug("Commands not found in hash, caching now...")
+                        self._cache_actions(data, config_data)
+                        self.get_actions(data)
+                        return
+                except subprocess.CalledProcessError, e:
+                    self.logger.error(str(e))
+                    self.host.reply(
+                        dict(
+                            err="Shotgun Desktop failed to get engine commands.",
+                            retcode=self.CACHING_ERROR,
+                            out="Caching failed!",
+                        ),
+                    )
+                    return
+
+        self.host.reply(
+            dict(
+                err="",
+                retcode=self.SUCCESSFUL_LOOKUP,
+                actions=all_actions,
+                pcs=pc_names,
+            ),
+        )
+
+    ###########################################################################
+    # Context managers
+
+    @contextlib.contextmanager
+    def _db_connect(self):
+        """
+        Context manager that initializes a DB connection on enter and
+        disconnects on exit.
+        """
+        connection = self._init_db()
+        cursor = connection.cursor()
+        yield (connection, cursor)
+        cursor.close()
+        connection.close()
+
+    ###########################################################################
+    # Internal methods
+
+    def _cache_actions(self, data, config_data):
+        """
+        Triggers the caching or recaching of engine commands.
+
+        :param dict data: The data passed down from the wss client.
+        :param dict config_data: A dictionary keyed by PipelineConfiguration
+            id containing a dict that contains, at a minimum, "lookup_hash"
+            and "contents_hash" keys.
+        """
+        self.logger.debug("Caching engine commands...")
+
+        script = os.path.join(
+            os.path.dirname(__file__),
+            "scripts",
+            "cache_commands.py"
+        )
+
+        self.logger.debug("Executing script: %s" % script)
+        hash_data = dict()
+
+        for pc_id, pc_data in config_data.iteritems():
+            hash_data[pc_id] = dict(
+                lookup_hash=config_data[pc_id]["lookup_hash"],
+                contents_hash=config_data[pc_id]["contents_hash"],
+            )
+
+        args_file = self._get_arguments_file(
+            dict(
+                cache_file=self._cache_path,
+                data=data,
+                sys_path=sys.path,
+                base_configuration=self.BASE_CONFIG_URI,
+                hash_data=hash_data,
+                engine_name=self.ENGINE_NAME,
+            )
+        )
+
+        output = None
+
+        try:
+            output = subprocess.check_output([sys.executable, script, args_file])
```

Comment:
Are the potential problems involved related to the parsability of the output caught and returned by check_output? If so, since I'm not parsing the output in any way, aren't the hoops that the v1 api is jumping through unnecessary here? I'm just wanting the output in case I need to pass it to a logger.error call. What the actual output is that's caught is of little consequence.